### PR TITLE
add new config-js structure

### PR DESCRIPTION
**Pull request by spasovski**
I had something more complicated locally that also moves much of the helper methods in store.js like `getField() getFieldValues() isField()` but decided against it until for now until we're certain of the direction.

### CODE REVIEW

File: src/config/firefox-desktop.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  },
+  versions: {
+    title: 'Versions',
+    key: 'version',
+    type: 'multi',
+    skipValidation: true,
+    values: [
+      { key: 'nightly', label: 'Nightly' },
+      { key: 'beta', label: 'Beta' },
+      { key: 'release', label: 'release' },
```
Comment (robhudson) [inline src/config/firefox-desktop.js]: These aren't versions.

File: src/state/store.js [hunk 2]
```diff
@@ -160,31 +162,38 @@ export const hasDefaultControlFields = derived(store, ($store) =>
...
+    const params = getParamsForQueryString(obj);
+    delete params.timeHorizon;
+    delete params.proportionMetricType;
+    delete params.activeBuckets;
+    delete params.visiblePercentiles;
+    params.probe = obj.probeName;
+    params.os = osValue;
+    params.channel = channelValue;
+    params.process = process;
+    params.versions = []; // TODO: This should not be expected by the API.
```
Comment (spasovski) [inline src/state/store.js]: @robhudson we found that this param is still expected by the API, probably something to flag for future removal

Comment (hamilton) [reply src/state/store.js]: @spasovski let's file an issue for this so we can track that separately.

**Review by hamilton — APPROVED** [review]
Generally looks good. We're not quite ready to generalize for Fenix yet but this is a great step in the right direction. Thanks @spasovski.

**Pull request merged**